core/src/main/java/me/aurium/beetle/api/task/sync/Synced.java

```diff
@@ -0,0 +1,11 @@
+package me.aurium.beetle.api.task.sync;
+
+/**
+ * something that can be compared if it is on the main thread or not.
+ * A248, advise me on naming ideas since Synced sounds fucking stupid
+ */
+public interface Synced {
```

Comment:
```suggestion
public interface ThreadContextDetection {
```
Could also be MainThreadDetection